cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/common/AbstractServiceRetryableMacroEvaluator.java

```diff
@@ -61,11 +61,36 @@ public String lookup(String property) throws InvalidMacroException {
   public String evaluate(String macroFunction, String... args) throws InvalidMacroException {
     if (!functionName.equals(macroFunction)) {
       // This shouldn't happen
-      throw new IllegalArgumentException("Invalid function name " + macroFunction
-                                           + ". Expecting " + functionName);
+      throw new IllegalArgumentException("Invalid function name " + macroFunction + ". Expecting " + functionName);
+    }
+
+    long delay = RETRY_BASE_DELAY_MILLIS;
```

Comment:
sure, I can.  Although there doesn't appear to be any unit tests for this particular class, so I'm a bit hesitant to change the behavior of the original function. 

Also, this is the second time the logic appears, I tend to veer on the side of copying instead of DRY until it appears at least three times.

Reply:
Sure, if we're optimizing for delivery time it might be ok to leave as-is and refactor later.

Reply:
ok. let me resolve this comment then.

Reply:
On a second thought, I think this actually should be a relatively low-risk change to refactor out since we're just moving it to a helper function. Can we go ahead and make that change in this PR?

Reply:
Hey Dennis, thanks for the suggestion.  It looks like the build has succeeded.

Per our conversation offline, is it okay to merge this PR as it is if I don't get to implementing this refactor as part of the current release?

Reply:
Sure, sounds good